### PR TITLE
(feat) add a new payment on IoT template

### PR DESCRIPTION
**Pull request by dselman**
Signed-off-by: Dan Selman <danscode@selman.org>

### CODE REVIEW

File: src/payment-upon-iot/grammar/template.tem [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Upon long button press, [{buyer}] shall pay to [{seller}] [{amountPerUnit}] for each short button press. A maximum of [{paymentCount}] payments may be made before this contract is COMPLETED.
```
Comment (jeromesimeon) [inline src/payment-upon-iot/grammar/template.tem]: I would be interested in getting a legal review on the text to know whether such a text could be legally enforceable.

File: src/payment-upon-iot/lib/logic.ergo [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+    return CounterResponse{
+		counter: state.counter,
+		paymentCount: state.paymentCount,
+	}
+  }
+
+	/**
+	 * When the contract is signed a payment obligation is emitted
+	 */
+  clause contractSigned(request : ContractSigned) : Response emits PaymentObligation {
```
Comment (jeromesimeon) [inline src/payment-upon-iot/lib/logic.ergo]: The clause does not emit obligations, should it? Otherwise, the `emits PaymentObligation` should be removed.

**Comment by dselman** [comment]
I've merged the latest changes from master and fixed the emit.

**Comment by dselman** [comment]
I've merged the latest changes from master and fixed the emit.

**Comment by dselman** [comment]
I've merged the latest changes from master and fixed the emit.

**Comment by dselman** [comment]
I've merged the latest changes from master and fixed the emit.

**Pull request merged**